doc/l2mc/IGMP_Snooping_HLD.md

```diff
@@ -0,0 +1,701 @@
+    querier                     = "true"/"false"                       ; Is the IGMP Snooping querier enabled  or disabled 
+    fast-leave                  = "true"/"false"                       ; Is IGMP fast leave enabled or disabled 
+    version                     = 1*DIGIT                   ; IGMP Snooping Version ( Version 1 or 2 or 3, Default: 2)
+    query-interval              = 5*DIGIT                   ; Query Interval Value (1 to 18000sec, Default:125sec)
+    last-member-query-interval  = 5*DIGIT                   ; Last Member Query Interval Value (100 to 25500ms; Default: 1000ms)
+    query-max-response-time     = 2*DIGIT                   ; Query Response Time (1 to 25sec, Default: 10sec)
+     
+#### CFG_L2MC_MROUTER_TABLE
+    ;Store Mrouter configuration
+    ;Status: work in progress
+    key                         = CFG_L2MC_MROUTER_TABLE|"Vlan"vlanid|mrouter_port_name       
+    ;field                      = value               
+
+####  CFG_L2MC_STATIC_MEMBER_TABLE
+    ;Store Static L2MC entry configuration per VLAN. 
+    ;Status: work in progress 
+    key                 =  CFG_L2MC_STATIC_MEMBER_TABLE|"vlan"vlanid|group_address|static_port_name
+    ;field              = value
+
+### 3.2.2 APP DB
+
+#### APP_L2MC_VLAN_TABLE
+	;Store Snooping state on VLAN
+    ;Status: work in progress
+    key         = APP_L2MC_VLAN_TABLE:"Vlan"vlanid
+    ;field      = value
+
+#### APP_L2MC_MEMBER_TABLE
+    ;Store L2MC member table entry.
+    ;Status: work in progress
+    key            = APP_L2MC_MEMBER_TABLE:"Vlan"vlanid:source_address:group_address:port_name
+    ;field         = value
+    type           = "static"/"dynamic"/"remote"           ; Static or Dynamic multicast group entry or remote entry learnt from MCLAG peer
+### APP_L2MC_MROUTER_TABLE
+    ;Store L2MC mrouter table entry.
+    ;Status: work in progress
+    key            = APP_L2MC_MROUTER_TABLE:"Vlan"vlanid:port_name
+    ;field         = value
+    type           = "static"/"dynamic"           ; Static or Dynamic multicast group entry
+
+### 3.2.3 STATE DB
+#### L2MC_STATE_MEMBER_TABLE
+    ;Store L2MC member state table entry for ICCPd communications.
+    ;Status: work in progress
+    key            = L2MC_STATE_MEMBER_TABLE|"Vlan"vlanid|source_address|group_address|port_name
+    ;field         = value
+    type           = "static"/"dynamic"           ; Static or Dynamic multicast group entry
+    
+   #### L2MC_STATE_MROUTER_TABLE
+    ;Store L2MC mrouter state table entry for ICCPd communications.
+    ;Status: work in progress
+    key            = L2MC_STATE_MROUTER_TABLE|"Vlan"vlanid|port_name
+    ;field         = value
+    type           = "static"/"dynamic"           ; Static or Dynamic mrouter entry
+    
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+    No changes
+
+## 3.3 Switch State Service Design
+### 3.3.1 L2mcOrch changes 
```

Comment:
Does this layer maintain infomation if this a new group entry or existing entry?

Reply:
Yes.